diff -rcN a/setup/srcconf.c b/setup/srcconf.c
*** a/setup/srcconf.c	2019-03-06 04:52:41.000000000 -0300
--- b/setup/srcconf.c	2020-03-06 16:10:26.920349003 -0300
***************
*** 962,968 ****
  #if defined(__SCO_VERSION__)
        fprintf (f, "CFLAGS=-O -D_KERNEL -D_DDI=8\n");
  #else
!       fprintf (f, "CFLAGS += -D_KERNEL\n");
  #endif
  #ifdef HAVE_KERNEL_FLAGS
        add_kernel_flags (f);
--- 962,968 ----
  #if defined(__SCO_VERSION__)
        fprintf (f, "CFLAGS=-O -D_KERNEL -D_DDI=8\n");
  #else
!       fprintf (f, "CFLAGS += -D_KERNEL -fno-pic\n");
  #endif
  #ifdef HAVE_KERNEL_FLAGS
        add_kernel_flags (f);
