diff -rcN a/setup/srcconf.c b/setup/srcconf.c
*** a/setup/srcconf.c	2019-03-06 04:52:41.000000000 -0300
--- b/setup/srcconf.c	2020-03-06 21:33:35.161559899 -0300
***************
*** 962,968 ****
  #if defined(__SCO_VERSION__)
        fprintf (f, "CFLAGS=-O -D_KERNEL -D_DDI=8\n");
  #else
!       fprintf (f, "CFLAGS += -D_KERNEL\n");
  #endif
  #ifdef HAVE_KERNEL_FLAGS
        add_kernel_flags (f);
--- 962,968 ----
  #if defined(__SCO_VERSION__)
        fprintf (f, "CFLAGS=-O -D_KERNEL -D_DDI=8\n");
  #else
!       fprintf (f, "CFLAGS += -D_KERNEL -fno-pic -fno-PIE\n");
  #endif
  #ifdef HAVE_KERNEL_FLAGS
        add_kernel_flags (f);
diff -rcN a/setup/srcconf_linux.inc b/setup/srcconf_linux.inc
*** a/setup/srcconf_linux.inc	2017-01-31 18:37:08.000000000 -0200
--- b/setup/srcconf_linux.inc	2020-03-06 21:35:41.338234404 -0300
***************
*** 51,70 ****
  {
  # if defined(__x86_64__)
    fprintf (f,
! 	     "CFLAGS += -O3 -fno-common -mcmodel=kernel  -mno-red-zone -fno-pie  -fno-asynchronous-unwind-tables -ffreestanding%s\n", fnsp);
  # else
  #   ifndef __arm__
    if (getenv ("NO_REGPARM") == NULL)
  	{
  	  fprintf (f,
! 		   "CFLAGS += -O3 -fno-common -ffreestanding -mregparm=3 -DUSE_REGPARM%s\n", fnsp);
  	}
    else
  	{
! 	  fprintf (f, "CFLAGS += -O3 -fno-common -ffreestanding -DNO_REGPARM%s\n", fnsp);
  	}
  #   else
!   fprintf (f, "CFLAGS += -O3 -fno-common -ffreestanding%s\n", fnsp);
  #   endif
  # endif
  }
--- 51,70 ----
  {
  # if defined(__x86_64__)
    fprintf (f,
! 	     "CFLAGS += -O2 -fno-common -mcmodel=kernel  -mno-red-zone -fno-pie  -fno-asynchronous-unwind-tables -ffreestanding%s\n", fnsp);
  # else
  #   ifndef __arm__
    if (getenv ("NO_REGPARM") == NULL)
  	{
  	  fprintf (f,
! 		   "CFLAGS += -O2 -fno-common -ffreestanding -mregparm=3 -DUSE_REGPARM%s\n", fnsp);
  	}
    else
  	{
! 	  fprintf (f, "CFLAGS += -O2 -fno-common -ffreestanding -DNO_REGPARM%s\n", fnsp);
  	}
  #   else
!   fprintf (f, "CFLAGS += -O2 -fno-common -ffreestanding%s\n", fnsp);
  #   endif
  # endif
  }
