diff --git a/lib/libOSSlib/Makefile b/lib/libOSSlib/Makefile
index 98029f1..a9bd175 100644
--- a/lib/libOSSlib/Makefile
+++ b/lib/libOSSlib/Makefile
@@ -1,6 +1,6 @@
 OBJS=libmain.o midiparser.o play_event.o
 INSTALLLIB=/lib
-LDFLAGS=-shared -fPIC
+LDFLAGS+=-shared -fPIC
 CFLAGS=-O -fPIC
 #LDFLAGS=-Bdynamic
 AROPTS=rc
diff --git a/lib/libsalsa/Makefile b/lib/libsalsa/Makefile
index 5f8216b..863ba7f 100644
--- a/lib/libsalsa/Makefile
+++ b/lib/libsalsa/Makefile
@@ -46,7 +46,7 @@ rawmidi.lo:	rawmidi.c
 		$(LIBTOOL) --mode=compile ${CC} $(CFLAGS) -c rawmidi.c 
 
 libsalsa.la:   $(LOBJECTS)
-		$(LIBTOOL) --mode=link ${CC} -rpath /usr/lib -Wl,--version-script=Versions -version-info 2:0:0 -o libsalsa.la $(LOBJECTS) -L../libOSSlib -lOSSlib
+		$(LIBTOOL) --mode=link ${CC} -rpath /usr/lib -Wl,--version-script=Versions -version-info 2:0:0 -o libsalsa.la $(LDFLAGS) $(LOBJECTS) -L../libOSSlib -lOSSlib
 
 install:	libsalsa.la
 		$(LIBTOOL) cp libsalsa.la /usr/lib
diff --git a/setup/srcconf.c b/setup/srcconf.c
index 0d37a16..6e20d54 100644
--- a/setup/srcconf.c
+++ b/setup/srcconf.c
@@ -928,7 +928,7 @@ printf("Symlink %s -> %s\n", source, target);
     fprintf (f, "CFLAGS=%s\n", conf.cflags);
 #endif
   if (*conf.ldflags != 0)
-    fprintf (f, "LDFLAGS=%s\n", conf.ldflags);
+    fprintf (f, "LDFLAGS+=%s\n", conf.ldflags);
 
   if (strcmp(conf.endianess, "UNKNOWN") != 0)
      sprintf (tmp_endian, " -DOSS_%s_ENDIAN", conf.endianess);
@@ -1170,7 +1170,7 @@ printf("Symlink %s -> %s\n", source, target);
 
       fprintf (f, "$(BINDIR)/%s:\t$(OBJECTS)\n", name);
       fprintf (f,
-	       "\t$(CC) $(CFLAGS) -s -o $(BINDIR)/%s $(OBJECTS) $(LIBRARIES) $(LDFLAGS) %s\n",
+	       "\t$(CC) $(CFLAGS) -o $(BINDIR)/%s $(OBJECTS) $(LIBRARIES) $(LDFLAGS) %s\n",
 	       name, extra_libraries);
       fprintf (f, "\n\n");
     }
@@ -1199,7 +1199,7 @@ printf("Symlink %s -> %s\n", source, target);
 
       fprintf (f, "$(SBINDIR)/%s:\t$(OBJECTS)\n", name);
       fprintf (f,
-	       "\t$(CC) $(CFLAGS) -s -o $(SBINDIR)/%s $(OBJECTS) $(LIBRARIES) $(LDFLAGS) %s\n",
+	       "\t$(CC) $(CFLAGS) -o $(SBINDIR)/%s $(OBJECTS) $(LIBRARIES) $(LDFLAGS) %s\n",
 	       name, extra_libraries);
       fprintf (f, "\n\n");
     }
@@ -1226,7 +1226,7 @@ printf("Symlink %s -> %s\n", source, target);
 		     obj_src[i], objdir, objects[i]);
 	  else
 	    fprintf (f,
-		     "\t$(CC) -c $(CFLAGS) $(OSFLAGS) $(LIBRARIES) $(INCLUDES) %s -o $(%s)/%s\n",
+		     "\t$(CC) -c $(CFLAGS) $(OSFLAGS) $(INCLUDES) %s -o $(%s)/%s $(LIBRARIES)\n",
 		     obj_src[i], objdir, objects[i]);
 	  fprintf (f, "\n");
 	}
