diff -rcN a/setup/Linux/oss/build/install.sh b/setup/Linux/oss/build/install.sh
*** a/setup/Linux/oss/build/install.sh	2014-02-14 20:59:55.000000000 -0200
--- b/setup/Linux/oss/build/install.sh	2020-03-06 13:48:29.820085653 -0300
***************
*** 179,184 ****
--- 179,186 ----
  fi
  
  cp -f ../objects/osscore.o osscore_mainline.o
+ rm -f osscore_lnk.c
+ ln -s osscore.c osscore_lnk.c
  
  rm -f Makefile
  ln -s Makefile.osscore Makefile
***************
*** 203,213 ****
  	exit 3
  fi
  
! if ! $LD -r osscore.ko osscore_mainline.o -o /lib/modules/$UNAME/kernel/oss/osscore.ko
! then
! 	echo Linking the osscore module failed
! 	exit 5
! fi
  
  if test -f Module.symvers
  then
--- 205,217 ----
  	exit 3
  fi
  
! #if ! $LD -r osscore.ko osscore_mainline.o -o /lib/modules/$UNAME/kernel/oss/osscore.ko
! #then
! #	echo Linking the osscore module failed
! #	exit 5
! #fi
! cp -f osscore.ko /lib/modules/$UNAME/kernel/oss/
! rm -f osscore_lnk.c
  
  if test -f Module.symvers
  then
***************
*** 228,237 ****
  	N=`basename $n .o`
  	echo Building module $N 
  
! 	rm -f $N_mainline.o Makefile
  
! 	sed "s/MODNAME/$N/" < Makefile.tmpl > Makefile
! 	ln -s $n $N_mainline.o
  
  	if ! make KERNELDIR=$KERNELDIR > build.list 2>&1
  	then
--- 232,245 ----
  	N=`basename $n .o`
  	echo Building module $N 
  
! 	#rm -f $N_mainline.o Makefile
! 	rm -f $N\_mainline.o  $N\_lnk.c Makefile
  
! 	#sed "s/MODNAME/$N/" < Makefile.tmpl > Makefile
! 	#ln -s $n $N_mainline.o
! 	sed "s/MODNAME/$N/g" < Makefile.tmpl > Makefile
! 	ln -s $N.c $N\_lnk.c
! 	ln -s $n $N\_mainline.o
  
  	if ! make KERNELDIR=$KERNELDIR > build.list 2>&1
  	then
***************
*** 240,252 ****
  		exit 4
  	fi
  
! 	if ! $LD -r $N.ko $N_mainline.o -o /lib/modules/$UNAME/kernel/oss/$N.ko
! 	then
! 		echo Linking $N module failed
! 		exit 6
! 	fi
! 
! 	rm -f $N_mainline.o
  	make clean
  done 
  
--- 248,263 ----
  		exit 4
  	fi
  
! 	#if ! $LD -r $N.ko $N_mainline.o -o /lib/modules/$UNAME/kernel/oss/$N.ko
! 	#then
! 	#	echo Linking $N module failed
! 	#	exit 6
! 	#fi
! 
! 	#rm -f $N_mainline.o
! 	cp -f $N.ko /lib/modules/$UNAME/kernel/oss/
! 	rm -f $N\_mainline.o
! 	rm -f $N\_lnk.c
  	make clean
  done 
  
diff -rcN a/setup/Linux/oss/build/Makefile.osscore b/setup/Linux/oss/build/Makefile.osscore
*** a/setup/Linux/oss/build/Makefile.osscore	2014-02-14 20:59:55.000000000 -0200
--- b/setup/Linux/oss/build/Makefile.osscore	2020-03-06 13:49:48.613419344 -0300
***************
*** 5,10 ****
--- 5,11 ----
  ifneq ($(KERNELRELEASE),)
  
  	obj-m := osscore.o
+ 	osscore-objs := osscore_lnk.o osscore_mainline.o
  
  else
  
diff -rcN a/setup/Linux/oss/build/Makefile.tmpl b/setup/Linux/oss/build/Makefile.tmpl
*** a/setup/Linux/oss/build/Makefile.tmpl	2014-02-14 20:59:55.000000000 -0200
--- b/setup/Linux/oss/build/Makefile.tmpl	2020-03-06 13:50:35.590086215 -0300
***************
*** 5,10 ****
--- 5,11 ----
  ifneq ($(KERNELRELEASE),)
  
  	obj-m := MODNAME.o
+ 	MODNAME-objs := MODNAME_lnk.o MODNAME_mainline.o
  
  else
  
