diff -rcN a/setup/Linux/oss/build/osscore.c b/setup/Linux/oss/build/osscore.c
*** a/setup/Linux/oss/build/osscore.c	2019-03-05 02:16:07.000000000 -0300
--- b/setup/Linux/oss/build/osscore.c	2020-03-06 22:20:45.608402999 -0300
***************
*** 32,37 ****
--- 32,38 ----
  #include <linux/sched.h>
  #include <linux/interrupt.h>
  #if LINUX_VERSION_CODE > KERNEL_VERSION(3,10,0)
+ #include <linux/cred.h>
  #include <linux/uidgid.h>
  #endif
  #if LINUX_VERSION_CODE >= KERNEL_VERSION(4,11,0)
***************
*** 478,484 ****
  unsigned int
  oss_get_uid (void)
  {
! #if LINUX_VERSION_CODE >= KERNEL_VERSION(3,10,0)
    return current->cred->uid.val;
  #else 
  #if LINUX_VERSION_CODE >= KERNEL_VERSION(3,14,0)
--- 479,488 ----
  unsigned int
  oss_get_uid (void)
  {
! #if LINUX_VERSION_CODE >= KERNEL_VERSION(3,13,0)
!   return current->cred->uid.val;
! #else 
! #if LINUX_VERSION_CODE >= KERNEL_VERSION(3,14,0)
    return current->cred->uid.val;
  #else 
  #if LINUX_VERSION_CODE >= KERNEL_VERSION(3,14,0)
***************
*** 489,494 ****
--- 493,499 ----
    return current->uid;
  #endif
  #endif
+ #endif
    return 0;
  }
  
